master/internal/experiment.go

```diff
@@ -252,412 +209,463 @@ func newUnmanagedExperiment(
 	telemetry.ReportExperimentCreated(expModel.ID, activeConfig)
 
 	// Will only have the model, nothing required for the experiment actor.
-	return &experiment{
+	return &internalExperiment{
 		Experiment: expModel,
 	}, nil, nil
 }
 
-func (e *experiment) Receive(ctx *actor.Context) error {
+// StartAsync first registers the experiment and then starts asynchronously.
+func (e *internalExperiment) StartAsync() (*errgroup.Group, error) {
```

Comment:
why do we think we can't safely move `StartAsync` into `new..` and always do it sync? or do we think we could do that as an immediate followup, at least?